include/seqan3/core/type_traits/basic.hpp

```diff
@@ -137,6 +138,33 @@ using ignore_t = remove_cvref_t<decltype(std::ignore)>;
 template <typename t>
 constexpr bool decays_to_ignore_v = std::is_same_v<remove_cvref_t<t>, ignore_t>;
 
+// ----------------------------------------------------------------------------
+// strip_type_identity
+// ----------------------------------------------------------------------------
+
+/*!\brief A transformation trait that wraps a type in std::type_identity unless it is already a std::type_identity.
+ * \implements seqan3::transformation_trait
+ * \tparam t The type to operate on.
+ */
+template <typename t>
+struct strip_type_identity : std::type_identity<t>
+{};
+
+/*!\brief A transformation trait that wraps a type in std::type_identity unless it is already a std::type_identity.
+ * \implements seqan3::transformation_trait
+ * \tparam t The type to operate on.
+ */
+template <typename t>
+struct strip_type_identity<std::type_identity<t>> : std::type_identity<t>
+{};
+
+/*!\brief A transformation trait shortcut that returns the type inside a std::type_identity or the type itself.
```

Comment:
You haven't changed the description here, is this a different case? Otherwise please adapt to the version above.

Reply:
No, it's the shortcut and the shortcut exposes the actual type. The transformation trait always is a std::type_identity. If I change the description here it will become very confusing I think.

Reply:
Ah, no then do not change it...